### PR TITLE
Refactor rollbar to use new routing api + catalog entity annotations

### PR DESCRIPTION
**Pull request by andrewthauer**
## Hey, I just made a Pull Request!

Refactored rollbar frontend to to use entities and the new catalog router:

- Added new annotation `rollbar.com/project-slug`
- Refactored frontend to use entities with `rollbar.com/project-slug` annotations
- Refactored frontend to use hooks
- Added `EntityPageRollbar` + `Router` to allow for creation of an `Errors` page for the catalog

NOTE: I have not integrated the router into the example app. This could be done similar to how #2282 is using a switcher or can be implemented by users following the instructions / pattern.

image

image

image

#### :heavy_check_mark: Checklist

- [ ] All tests are passing `yarn test`
- [ ] Screenshots attached (for UI changes)
- [ ] Relevant documentation updated
- [ ] Prettier run on changed files
- [ ] Tests added for new functionality
- [ ] Regression tests added for bug fixes

### CODE REVIEW

**Comment by freben** [comment]
Build currently fails due to the organization prop.

File: plugins/rollbar/src/components/Router.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import { Routes, Route } from 'react-router';
+import { Entity } from '@backstage/catalog-model';
+import { WarningPanel } from '@backstage/core';
+import { catalogRouteRef } from '../routes';
+import { ROLLBAR_ANNOTATION } from '../constants';
+import { EntityPageRollbar } from './EntityPageRollbar/EntityPageRollbar';
+
+export const isPluginApplicableToEntity = (entity: Entity) =>
+  Boolean(entity.metadata.annotations?.[ROLLBAR_ANNOTATION]) &&
+  entity.metadata.annotations?.[ROLLBAR_ANNOTATION] !== '';
```
[the remark this answers is not in the document]
Comment (freben) [reply plugins/rollbar/src/components/Router.tsx:27]: (i see that this was copied from elsewhere - no worries - we'll fix it in the other place too soon)

File: plugins/rollbar/src/components/Router.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+type Props = {
+  entity: Entity;
+};
+
+export const Router = ({ entity }: Props) =>
+  !isPluginApplicableToEntity(entity) ? (
+    <WarningPanel title="Rollbar plugin:">
+      `entity.metadata.annotations['{ROLLBAR_ANNOTATION}']` key is missing on
```
Comment (freben) [inline plugins/rollbar/src/components/Router.tsx:36]: replace backticks with `<pre>` perhaps

File: plugins/rollbar/src/hooks/useCatalogEntity.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  catalogApiRef,
+  useEntityCompoundName,
+} from '@backstage/plugin-catalog';
+
+export function useCatalogEntity() {
+  const catalogApi = useApi(catalogApiRef);
+  const { namespace, name } = useEntityCompoundName();
+
+  const { value: entity, error, loading } = useAsync(
+    () => catalogApi.getEntityByName({ kind: 'Component', namespace, name }),
```
Comment (freben) [inline plugins/rollbar/src/hooks/useCatalogEntity.ts:29]: @shmidt-i @Rugvip this code also relates to #1947 - we should settle on a way for the catalog wrapper to define a sensible root route format for the catalog (e.g. either `/catalog/default/component/my-component` or `/catalog/component%3Amy-component` or whatever) and then make sure there's a simple hook to get `{ kind, namespace, name }` and one to get `[entity, loading, error]`.

@andrewthauer no action on your part, this is fine in the current state of things

File: plugins/rollbar/src/hooks/useProject.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+import { useApi, configApiRef } from '@backstage/core';
+import { Entity } from '@backstage/catalog-model';
+import { ROLLBAR_ANNOTATION } from '../constants';
+
+export function useProjectSlugFromEntity(entity: Entity) {
+  const configApi = useApi(configApiRef);
+
+  const [orgOrProject, project] = (
+    entity?.metadata?.annotations?.[ROLLBAR_ANNOTATION] ?? ''
+  ).split('/');
```
Comment (freben) [inline plugins/rollbar/src/hooks/useProject.ts:26]: trick: `const [project, org] = <annotation>.split('/').reverse()` :) if you like

**Comment by andrewthauer** [comment]
@freben - Thanks! All relevant comments addressed and CI now passing.

**Review by freben — APPROVED** [review]

**Pull request merged**